test/e2e/set/scyllacluster/datainserter.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2021 ScyllaDB
+
+package scyllacluster
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/gocql/gocql"
+	"github.com/scylladb/gocqlx/v2"
+	"github.com/scylladb/gocqlx/v2/table"
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const nRows = 10
+
+type DataInserter struct {
+	client            corev1client.CoreV1Interface
+	session           *gocqlx.Session
+	keyspace          string
+	table             *table.Table
+	data              []*TestData
+	replicationFactor int
+}
+
+type TestData struct {
+	Id   int    `db:"id"`
+	Data string `db:"data"`
+}
+
+func NewDataInserter(ctx context.Context, client corev1client.CoreV1Interface, sc *scyllav1.ScyllaCluster, replicationFactor int) (*DataInserter, error) {
+	if replicationFactor < 1 {
+		return nil, fmt.Errorf("replication factor can't be set to less than 1")
+	}
+
+	keyspace := strings.Replace(string(uuid.NewUUID()), "-", "", -1)
+	d := &DataInserter{
+		client:   client,
+		keyspace: keyspace,
+		table: table.New(table.Metadata{
+			Name:    fmt.Sprintf(`"%s"."test"`, keyspace),
+			Columns: []string{"id", "data"},
+			PartKey: []string{"id"},
+		}),
+		data: func() []*TestData {
+			lst := make([]*TestData, nRows)
+			for i := 0; i < nRows; i++ {
+				lst[i] = &TestData{Id: i, Data: rand.String(32)}
+			}
+			return lst
+		}(),
+		replicationFactor: replicationFactor,
+	}
+
+	session, err := d.newSession(ctx, sc)
```

Comment:
nit: you could create the session before the object and use initialization to set it

Reply:
Unless there is a different way to do this that I'm not aware of, I'd have to revert the change requested in [this comment](https://github.com/scylladb/scylla-operator/pull/692#discussion_r676734559), but I believe it's better we keep it as a member method?

Reply:
I meant to do it just before `di := &DataInserter{` but assigning into `di.session` works too